### ISSUE TITLE
Problem in setting up newww: npm owner ls pngcrush-bin

### ISSUE THREAD

**Issue by icyflame** [npm/newww#1015]
I followed the instructions given at the end of the README file to setup the dev server on my Ubuntu machine. I got the following error:

![image](https://cloud.githubusercontent.com/assets/3668034/7898093/cdc617d6-0712-11e5-87fa-57f3ec62304e.png)

I found that #766 was this same issue, but it was closed by the author without any explanation as to what solved the problem.

Please help me on this.

**Comment by alrra** [comment]
@icyflame 

> Failed at the pngcrush-bin@1.0.0 postinstall script 'node lib/install.js'
> This is most likely a problem with pngcrush-bin package.

Since the specified [`devDependencies`](https://github.com/npm/newww/blob/dcfcf5b03736d0a07d3ad1664f48abe1f6fab717/package.json#L96-L119) are quite outdated, try updating [`imagemin-pngcrush`](https://github.com/npm/newww/blob/e869f8df71de044e44ce3dbbbb0a4e4a45f9457a/package.json#L111) to the latest version (currently `4.1.0`), and see if that works. If it does not, it means that there is a problem with [`pngcrush-bin`](https://github.com/imagemin/pngcrush-bin), and thus, you should open an issue in its [repository](https://github.com/imagemin/pngcrush-bin).

--

Note: I works for me on OS X 10.10.3.

**Comment by icyflame** [comment]
@alrra 

Thanks a lot!

I changed it in the devDependencies and it worked!

But then, when I started the server I ended up with an error that was reported just before mine.

I ended up with the same problem as in https://github.com/npm/newww/issues/1008

This is my server log tail, after the Hapi server has started:

```shell
[23:13:27] Starting 'fonts'...
[23:13:27] Finished 'fonts' after 46 ms
[23:13:27] Starting 'images'...
[23:13:27] Finished 'images' after 2.1 ms
[23:13:27] Starting 'misc'...
[23:13:27] Finished 'misc' after 1.17 ms
[23:13:27] Starting 'styles'...
[23:13:27] Finished 'styles' after 5.84 ms
[23:13:27] Starting 'browserify'...
[23:13:27] Finished 'browserify' after 34 ms
[23:13:27] Starting 'concat'...
[23:13:27] Finished 'concat' after 3.58 ms
[23:13:27] Starting 'build'...
[23:13:27] Finished 'build' after 9.28 μs
[23:13:27] Starting 'nodemon'...
[23:13:27] Finished 'nodemon' after 2.05 ms
[23:13:27] Starting 'watch'...
[23:13:27] Finished 'watch' after 533 ms
[23:13:27] Starting 'dev'...
[23:13:27] Finished 'dev' after 11 μs
[gulp] [nodemon] v1.3.7
[gulp] [nodemon] to restart at any time, enter `rs`
[gulp] [nodemon] watching: *.*
[gulp] [nodemon] starting `node server.js`
2015-05-30T17:45:00.097Z  info server: server repl socket at /tmp/rpl/www-undefined.sock
2015-05-30T17:45:38.324Z  info server: Hapi server started @ https://localhost:15443
2015-05-30T17:46:12.193Z error EXTERNAL: {
  "time": "2015-05-30T17:46:12.193Z",
  "hostname": "siddharth-inspiron-5547",
  "pid": 27615,
  "level": "error",
  "name": "EXTERNAL"
}
2015-05-30T17:46:12.193Z error EXTERNAL: Error: connect ECONNREFUSED
  at errnoException (net.js:904:11)
  at Object.afterConnect [as oncomplete] (net.js:895:19)

Unhandled rejection Error: connect ECONNREFUSED
  at errnoException (net.js:904:11)
  at Object.afterConnect [as oncomplete] (net.js:895:19)

2015-05-30T17:46:12.200Z error EXTERNAL: {
  "time": "2015-05-30T17:46:12.200Z",
  "hostname": "siddharth-inspiron-5547",
  "pid": 27615,
  "level": "error",
  "name": "EXTERNAL"
}
2015-05-30T17:46:12.200Z error EXTERNAL: Error: connect ECONNREFUSED
  at errnoException (net.js:904:11)
  at Object.afterConnect [as oncomplete] (net.js:895:19)

2015-05-30T17:46:12.208Z error EXTERNAL: {
  "time": "2015-05-30T17:46:12.208Z",
  "hostname": "siddharth-inspiron-5547",
  "pid": 27615,
  "level": "error",
  "name": "EXTERNAL"
}
2015-05-30T17:46:12.208Z error EXTERNAL: Error: connect ECONNREFUSED
  at errnoException (net.js:904:11)
  at Object.afterConnect [as oncomplete] (net.js:895:19)

2015-05-30T17:46:13.179Z  info EXTERNAL: - https://localhost:15443 - [30/May/2015:23:16:13 +0530] "GET https://api.npmjs.org/downloads/point/last-week HTTP/1.1" 200 -
2015-05-30T17:46:13.183Z  info EXTERNAL: - https://localhost:15443 - [30/May/2015:23:16:13 +0530] "GET https://api.npmjs.org/downloads/point/last-month HTTP/1.1" 200 -
2015-05-30T17:46:13.185Z  info EXTERNAL: - https://localhost:15443 - [30/May/2015:23:16:13 +0530] "GET https://api.npmjs.org/downloads/point/last-day HTTP/1.1" 200 -

```
Do you have any idea as to why this issue is being thrown? And whether it is related to:

> Because this new registry2 architecture uses lots of separate microservices, it can be cumbersome to configure a development environment. To simplify development setup, the r2 branch no longer uses the vagrant image, but instead points to various services in a remote staging environment. As of this writing, that environment is only accessible to npm employees, but we will soon offer an alternative that allows the public to more easily bootstrap a newww development environment.

as written by @zeke at https://github.com/npm/newww/issues/761 ?

is the remote staging environment refusing connections? (I have already ensured that the port that this process is using is available.)

**Comment by alrra** [comment]
> I changed it in the devDependencies and it worked!

@icyflame I've open #1016 to start the discussion on updating the `dependencies` and `devDependencies`, so maybe this should be close in favor of #1016? (Cc: @ceejbot, @rockbot).

--

> I ended up with the same problem as in #1008

@icyflame Yes, I have the same issue.

**Comment by icyflame** [comment]
@alrra 
Yes, since you have the same issue as mentioned in #1008 , the problem that this issue was opened with can be fixed by merging #1016 (changing `devDependencies` version to the latest version)

Closing this in favour of merge of #1016
